Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -1122,9 +1122,36 @@ int Element::scroll_width() const
     return 0;
 }
 
+// https://w3c.github.io/csswg-drafts/cssom-view/#dom-element-scrollheight
 int Element::scroll_height() const
 {
-    dbgln("FIXME: Implement Element::scroll_height() (called on element: {})", debug_description());
+    // 1. Let document be the element’s node document.
+    auto& document = this->document();
+
+    // 2. If document is not the active document, return zero and terminate these steps.
+    if (!document.is_active())
+        return 0;
+
+    // 3. Let viewport height be the height of the viewport excluding the height of the scroll bar, if any, or zero if there is no viewport.
+    auto viewport_height = document.browsing_context()->viewport_rect().height().value();
+    if (paint_box()) {
+        viewport_height -= paint_box()->absolute_padding_box_rect().height().value();
+    }
+
+    // 4. If the element is the root element and document is not in quirks mode return max(viewport scrolling area height, viewport height).
+    if (document.document_element() == this && !document.in_quirks_mode())
+        return AK::max(document.browsing_context()->viewport_rect().height().value(), viewport_height);
```

Comment:
The height of the viewport rect is not the height of the viewport scrolling area. My understanding is, the "scrolling area" is the size of the content inside the viewport. See the "rightward and downward"/"For a viewport" box in this table: https://w3c.github.io/csswg-drafts/cssom-view/#scrolling-area